src/libraries/Microsoft.PowerFx.Interpreter/Functions/CollectFunction.cs

```diff
@@ -0,0 +1,362 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Functions.FunctionArgValidators;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+using static Microsoft.PowerFx.Core.Localization.TexlStrings;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    // The CollectFunction class was inspired (copied) by Set function.
+    // Implementation of a Set function which just chains to 
+    // RecalcEngine.UpdateVariable().
+    // Set has no return value. 
+    // Whereas PowerApps' Set() will implicitly define arg0,
+    //  this Set() requires arg0 was already defined and has a type.
+    //
+    // Called as:
+    //   Set(var,newValue)
+    internal class CollectFunction : BuiltinFunction, IAsyncTexlFunction
```

Comment:
>CollectFunction [](http://example.com/codeflow?start=19&length=15)

this is copied from PowerApps?

It would be good to run under code-coverage and see:
- if there are sections here we can remove 
- new tests we should add around collect?

Reply:
It was copied from AP. The not covered code is currently at 47.23%. More test cases will be added as needed.